hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/keygen/BuiltinKeyGenerator.java

```diff
@@ -81,7 +99,18 @@ public String getPartitionPath(Row row) {
     return getKey(genericRecord).getPartitionPath();
   }
 
-  void buildFieldPositionMapIfNeeded(StructType structType) {
+  /**
+   * Fetch partition path from {@link InternalRow}.
+   *
+   * @param internalRow {@link InternalRow} instance from which partition path needs to be fetched from.
+   * @param structType  schema of the internalRow.
+   * @return the partition path.
+   */
+  public String getPartitionPath(InternalRow internalRow, StructType structType) {
+    throw new UnsupportedOperationException("Operation not supported. Please override if required.");
```

Comment:
Yet to figure out a way to fix this. Will update the patch once I have the solution. But atleast for all built in key gen, have concrete impls.